lib/email.js

```diff
@@ -39,19 +49,72 @@ api.process = function(options, callback) {
         brMessagesPush.queue.pull(o, callback);
       },
       collect: ['pull', function(callback, results) {
-        console.log('PULLRESULTS', results.pull.value);
-        if(!results.pull.value) {
+        if(!results.pull) {
+          done = true;
+          return callback();
+        }
+        brMessages.getMessages(
+          null,
+          results.pull.messages,
+          {recipient: results.pull.recipient},
+          callback
+        );
+      }],
+      identity: ['pull', function(callback, results) {
+        if(!results.pull) {
+          done = true;
+          return callback();
+        }
+        brIdentity.get(null, results.pull.recipient, function(err, identity) {
+          callback(err, identity);
+        });
+      }],
+      email: ['collect', 'identity', function(callback, results) {
+        if(!results.collect || !results.identity) {
           done = true;
           return callback();
         }
-        // TODO: pull together details needed to actually generate an email
-        // need to get email address, message contents, etc.
-        // need to build a single message from all the messages listed in messages array
+        // TODO: Add ability to send out different batches of email
+        // based on message type.
+        bedrock.events.emitLater({
+          type: messageEmailEventType,
+          details: {
+            messages: results.collect,
+            identity: results.identity
+          }
+        });
+
         callback();
       }],
-      remove: ['collect', function(callback, results) {
-        brMessagesPush.queue.remove({jobId: jobId}, callback);
+      remove: ['email', function(callback, results) {
+        brMessagesPush.queue.remove(jobId, callback);
       }]
-    }, callback);
+    }, function(err, results) {
+      if(err) {
+        // If error, jobs will remain in the queue to be
+        // completed during another cycle.
+        if(err.name === 'NotFound') {
+          // Only exception is if Identity no longer exists,
+          // then remove the job from the queue and continue
+          logger.debug('Removing email job' + jobId +
+                      ' from queue because its recipient no longer exists');
+          return brMessagesPush.queue.remove(jobId, callback);
+        }
+      }
+      callback(err, results);
+    });
   }, callback);
 };
+
+api.getOptions = function() {
+  return {
+    label: 'Email',
+    type: 'email',
+    intervals: [
+      {label: 'Daily', value: 'daily'},
+      {label: 'Immediate', value: 'immediate'}
```

Comment:
This approach doesn't support internationalization ... we should put a TODO here to figure that out later.